contentrouting/interface.go

```diff
@@ -3,23 +3,51 @@ package contentrouting
 import (
 	"context"
 
-	"github.com/multiformats/go-multicodec"
 	cid "github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multicodec"
 )
 
 // The Routing interface is an evolution of the ContentRouting interface
 // found in [libp2p](https://github.com/libp2p/go-libp2p-core/blob/master/routing/routing.go#L26).
 //
 // Routing focuses only on the retrieval half of the interface: how do
 // I located content given a content identifier?
-interface Routing {
+type Routing interface {
 	FindProvidersAsync(context.Context, cid.Cid) <-chan RoutingRecord
```

Comment:
Just to be safe, I'd add a varargs in the parameters.... just call it a ContentRoutingOption with a blank interface for now?

Given we are returning a channel, we should specify what happens if the consumer fails to consume it. (i.e. does our system get stuck? Can we keep processing effectively?)